<?xml version="1.0"?>

<launch>



<!--
    <group ns="lumi_mujoco">
        <rosparam command="load" file="$(find lumi_mujoco)/config/controllers.yaml"/>
        <arg name="controllers" default="joint_state_controller trajectory_controller gripper_controller"/>
        <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false" output="screen"
              args="$(arg controllers)"/>

        <node pkg="topic_tools" type="relay" name="relay_joint_state" args="joint_states /joint_states"/>
    </group>

    <node name="model" pkg="mujoco_ros_control" type="mujoco_control" output="screen">
        <param name="doormodel" value="$(arg mujoco_model)"/>
        <param name="look_at_x" value="$(arg look_at_x)"/>
        <param name="look_at_y" value="$(arg look_at_y)"/>
        <param name="look_at_z" value="$(arg look_at_z)"/>
        <remap from="/lumi_mujoco/trajectory_controller/follow_joint_trajectory" to="/lumi_arm_controller/follow_joint_trajectory"/>
        <remap from="/lumi_mujoco/gripper_controller/gripper_cmd" to="/lumi_hand_controller/gripper_action"/>
        <param name="lumi_joint4_init" value="-1.5708"/>
        <param name="lumi_joint6_init" value="1.5708"/>
    </node>    

    <include file="$(find lumi_moveit_config)/launch/lumi_moveit.launch">
        <arg name="rviz" value="$(arg rviz)"/>
    </include>
-->
    <arg name="urdf_door" default="$(find simulator)/test.urdf"/>
    <arg name="model" default="$(find simulator)/samplefinal.xml"/>


    <include file="$(find lumi_mujoco)/launch/simulation.launch">
        <!--<arg name="model" default="$(find lumi_mujoco)/launch/samplefinal.xml"/>-->
        <arg name="model" default="$(arg model)"/>
        <arg name="mujoco_model" default="$(arg model)"/>

        <!--<arg name="model" value="$(arg urdf_door)"/>-->

        <!--<arg name="mujoco_model" default="$(arg model)"/>-->
        <!--<arg name="urdf_table" default="$(find lumi_description)/robots/robot_table_env.urdf"/>
        <arg name="xml_table" default="$(find lumi_description)/robots/robot_table_env.xml"/>-->
        
        <arg name="urdf_model" default="$(arg urdf_door)"/>


        <!--<arg name="rviz" default="true"/>-->
        <arg name="look_at_x" default="5"/>
        <arg name="look_at_y" default="5"/>
        <arg name="look_at_z" default="2"/>
    </include>


</launch>
